GpxLocationManager/LocationManager.swift

```diff
@@ -22,202 +22,195 @@ open class LocationManager {
     }
   }
 
-  open func authorizationStatus() -> CLAuthorizationStatus {
+  internal func on<Result>(gpx: (GpxLocationManager) -> Result, core: (CLLocationManager) -> Result) -> Result {
     switch locationManagerType {
     case .gpxFile, .locations:
-      return GpxLocationManager.authorizationStatus()
+        return gpx(self.gpxLocationManager)
     case .coreLocation:
-      return CLLocationManager.authorizationStatus()
+        return core(self.cLLocationManager)
     }
   }
 
-  open var location: CLLocation? {
+  internal func on(gpx: ((GpxLocationManager) -> Void)?, core: ((CLLocationManager) -> Void)?) {
     switch locationManagerType {
     case .gpxFile, .locations:
-        return gpxLocationManager.location
+        gpx?(self.gpxLocationManager)
     case .coreLocation:
-        return cLLocationManager.location
+        core?(self.cLLocationManager)
     }
   }
 
+  open func authorizationStatus() -> CLAuthorizationStatus {
+    return self.on(gpx: { (_) in
+        return GpxLocationManager.authorizationStatus()
+    }, core: { (_) in
+        return CLLocationManager.authorizationStatus()
+    })
+  }
+
+  open var location: CLLocation? {
+    return self.on(gpx: { (manager) in
```

Comment:
Please remove the parens around `manager`. Per the Ray Wenderlich style [guide](https://github.com/raywenderlich/swift-style-guide#closure-expressions), which we should mention in the contributing document, we do not use parens in this situation.